### DISCUSSION TITLE
add custom snippets from github with git-ssh url/auth

### DISCUSSION THREAD

**Discussion by armyers** [LazyVim/LazyVim#2128] [Q&A] [answered]
Hi,

I was able to follow the example code to load custom snippets from a local file, and I'm able to create and use a github public repo with custom snippets. But I'm not able to load custom snippets from a repo in our private github org (for work). We're using ssh creds so the clone command is `git clone git@github.com:SomeOrg/local-snippets.git`. I've tried this in `~/.config/nvim/lua/plugins/luasnip.lua`:

```lua
return {
  "L3MON4D3/LuaSnip",
  dependencies = {
    { "SomeOrg/local-snippets", url= "git@github.com:SomeOrg/local-snippets.git" },
    config = function()
      require("luasnip.loaders.from_vscode").lazy_load()
    end,
  },
}
```

but I'm getting loading errors from lazy.

```
Failed to load plugins.luasnip

...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/plugin.lua:67: attempt to call method 'find' (a nil value)

# stacktrace:
  - ~/.config/nvim/lua/config/lazy.lua:9
  - ~/.config/nvim/init.lua:2
```

NOTE: I edited the above error messages.

thanks for your assistance!

**Comment by dpetka2001** [comment, marked as answer]
I believe the `config` function should be inside the curly brackets of `SomeOrg/local-snippets` after the `url` option.

**Comment by armyers** [comment]
that works!

```lua
return {
  "L3MON4D3/LuaSnip",
  dependencies = {
    {
      "SomeOrg/local-snippets",
      url = "git@github.com:SomeOrg/local-snippets.git",
      config = function()
        require("luasnip.loaders.from_vscode").lazy_load()
      end,
    },
  },
}
```

thank you again! :)
